### PR TITLE
[libra-framework] Reorganized prologue checks

### LINKED ISSUE #6321: [Bug] submit should return transaction expired error if the transaction is expired
# 🐛 Bug

## To reproduce

Example signed transaction hex: 1668F6BE25668C1A17CD8CAF6B8D2F2502000000000000000160A11CEB0B0100000006010002030206040802050A0707111A082B100000000100010101000201060C000109000C4C696272614163636F756E740C6164645F63757272656E63790000000000000000000000000000000101010001030B0038000201070000000000000000000000000000000105436F696E3105436F696E31000010270000000000000100000000000000034C425200CA9A3B00000000020020F549A91FB9989883FB4D38B463308F3EA82074FB39EA74DAE61F62E11BF55D254086A0D142FE8397A2C27888F007C4A920E28A1411CCDE9D7E9E2A01CAAFAD0271A8EE56F82F5940FC7A556434E5CDEAB776E8B00FBA9F9721642D9A97E25F0A07

The transaction expiration timestamp is set to 1000000000  (2001-09-08 18:46:40 -0700)
Submit success.
The transaction is not executed and drop some where unknown.

## Expected Behavior

Submit should return MempoolStatusCode::VmError error code
MempoolStatusCode has VmError: transaction didn't pass vm_validation
and vm validation has: TRANSACTION_EXPIRED

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Swapped the order of two checks to fix #6321 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Will add tests in e2e test shortly.

### CODE REVIEW

File: language/e2e-testsuite/src/tests/verify_txn.rs [hunk 1]
```diff
@@ -479,6 +479,23 @@ fn verify_expiration_time() {
         executor.execute_transaction(txn).status(),
         StatusCode::TRANSACTION_EXPIRED
     );
+
+    let txn = transaction_test_helpers::get_test_signed_transaction(
```
Comment (sblackshear) [inline language/e2e-testsuite/src/tests/verify_txn.rs:483]: Could you add a comment explaining that the 10 is deliberately chosen to try to trigger `SEQUENCE_NUMBER_TOO_NEW`?

**Review by sblackshear — COMMENTED** [review]
This feels reasonable to commit to me. Having the expiration check after the `SEQUENCE_NUMBER_TOO_NEW_CHECK` is a bug (though a minor one) and this fix is straightforward.

**Comment by xli** [comment]
could you add a test to json-rpc/tests/integration_test.rs , submit an expired transaction with large seq number, and assert submit returns error? you can checkout "invalid transaction submitted: mempool validation error" for example.

**Comment by runtian-zhou** [comment]
Added the fix per @xli's request

**Review by sblackshear — APPROVED** [review]

**Review by xli — APPROVED** [review]

File: json-rpc/tests/integration_test.rs [hunk 2]
```diff
@@ -372,6 +372,45 @@ fn create_test_cases() -> Vec<Test> {
...
+            run: |env: &mut testing::Env| {
+                env.allow_execution_failures(|env: &mut testing::Env| {
+                    let txn1 = {
+                        let account1 = env.get_account(0, 0);
+                        let account2 = env.get_account(1, 0);
+                        let script = transaction_builder_generated::stdlib::encode_peer_to_peer_with_metadata_script(
+                            coin1_tmp_tag(),
+                            account2.address,
+                            100,
+                            // todo: add metadata
```
Comment (xli) [inline json-rpc/tests/integration_test.rs:386]: may remove this todo

File: json-rpc/tests/integration_test.rs [hunk 3]
```diff
@@ -372,6 +372,45 @@ fn create_test_cases() -> Vec<Test> {
...
+                            vec![],
+                        );
+                        let seq = env
+                            .get_account_sequence(account1.address.to_string())
+                            .expect("account should exist onchain for create transaction");
+                        libra_types::transaction::helpers::create_user_txn(
+                            account1,
+                            TransactionPayload::Script(script),
+                            account1.address,
+                            // An expired transaction with too new sequence number will still be rejected
```
Comment (xli) [inline json-rpc/tests/integration_test.rs:397]: may use this comment as test name

**Pull request closed without merging**